devito/core/gpu_openacc.py

```diff
@@ -1,13 +1,20 @@
-from functools import partial
+from functools import partial, singledispatch
 
 import cgen as c
+from sympy import Function
 
 from devito.core.gpu_openmp import (DeviceOpenMPNoopOperator, DeviceOpenMPIteration,
```

Comment:
Looks a little odd to import from gpu_openmp for gpu_openacc. Might want to refine abstraction in the future.

Reply:
There is a TODO a few lines below which says:

> TODO: currently inhereting from the OpenMP Operators. Ideally, we should/could
abstract things away so as to have a separate, language-agnostic superclass